src/app/qgsidentifyresultsdialog.cpp

```diff
@@ -531,19 +532,83 @@ void QgsIdentifyResultsDialog::addFeature( QgsVectorLayer *vlayer, const QgsFeat
     connect( vlayer, &QgsVectorLayer::editingStopped, this, &QgsIdentifyResultsDialog::editingToggled );
   }
 
-  QgsIdentifyResultsFeatureItem *featItem = new QgsIdentifyResultsFeatureItem( vlayer->fields(), f, vlayer->crs() );
-  featItem->setData( 0, Qt::UserRole, FID_TO_STRING( f.id() ) );
+  QgsIdentifyResultsFeatureItem *featItem = createFeatureItem( vlayer, f, derivedAttributes, true, layItem );
```

Comment:
Could this be a `std::unique_ptr`?

Reply:
It doesn't have ownership - createFeatureItem automatically adds the item to the parent item (by necessity, because some parts of that code access the item's view)